csharp/src/iceboxnet/ServiceManager.cs

```diff
@@ -606,34 +549,25 @@ private void StopAll()
                 }
 
                 _services.Clear();
-                ServicesStopped(stoppedServices, _observers.Keys);
+                ServicesStopped(stoppedServices, _observers);
             }
         }
 
-        private void ServicesStarted(List<string> services, Dictionary<IServiceObserverPrx, bool>.KeyCollection observers)
+        private void ServiceStarted(string service, IEnumerable<IServiceObserverPrx> observers)
```

Comment:
Why does _mutex need to be locked? Could this method be static? Should we make it async?

Reply:
I think the lock is required because the observers list can change while iterating over it